codebuild/bin/s2n_setup_env.sh

```diff
@@ -122,6 +122,10 @@ fi
 # Select the libcrypto to build s2n against. If this is unset, default to the latest stable version(Openssl 1.1.1)
 if [[ -z $S2N_LIBCRYPTO ]]; then export LIBCRYPTO_ROOT=$OPENSSL_1_1_1_INSTALL_DIR ; fi
 if [[ "$S2N_LIBCRYPTO" == "openssl-1.1.1" ]]; then export LIBCRYPTO_ROOT=$OPENSSL_1_1_1_INSTALL_DIR ; fi
+if [[ "$S2N_LIBCRYPTO" == "openssl-3.0" ]]; then
+    export LIBCRYPTO_ROOT=$OPENSSL_3_0_INSTALL_DIR ;
+    export OPENSSL_CONF=$(realpath codebuild/openssl.config) ;
+fi
```

Comment:
Will this work for all customer builds, not just the CI? Can we do this from cmake / make / the actual C code?

Reply:
discussed offline.
cmake/make will cause dupe, while C code solution will require `ifdef`.

In this case we are going with the C code option since its easy for things to get lost in our build settings and loading the "legacy" provider is necessary for the code to work.